src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -40,46 +48,45 @@ public class WebSocketsTransport : ITransport
             }).Unwrap();
         }
 
-        public async Task ReceiveMessages(Uri pollUrl, CancellationToken cancellationToken)
+        private async Task ReceiveMessages(Uri pollUrl, CancellationToken cancellationToken)
         {
-            var totalBytes = 0;
-            var incomingMessage = new List<byte[]>();
             while (!cancellationToken.IsCancellationRequested)
             {
-                bool completedMessage;
+                const int bufferSize = 1024;
+                var totalBytes = 0;
+                var incomingMessage = new List<byte[]>();
                 WebSocketReceiveResult receiveResult;
-                ArraySegment<byte> buffer;
                 do
                 {
-                    buffer = new ArraySegment<byte>(new byte[1024]);
+                    var buffer = new ArraySegment<byte>(new byte[bufferSize]);
                     receiveResult = await _webSocket.ReceiveAsync(buffer, cancellationToken);
-                    completedMessage = receiveResult.EndOfMessage;
                     incomingMessage.Add(buffer.Array);
-                    totalBytes += buffer.Count;
-                } while (!completedMessage);
+                    totalBytes += receiveResult.Count;
+                } while (!receiveResult.EndOfMessage);
 
                 Message message;
                 if (incomingMessage.Count > 1)
                 {
                     var totalBuffer = new byte[totalBytes];
                     var offset = 0;
-                    for (int i = 0 ; i < incomingMessage.Count; i++)
+                    for (var i = 0 ; i < incomingMessage.Count; i++)
                     {
-                        System.Buffer.BlockCopy(incomingMessage[i], 0, totalBuffer, offset, incomingMessage[i].Length);
+                        Buffer.BlockCopy(incomingMessage[i], 0, totalBuffer, offset, incomingMessage[i].Length);
                         offset += incomingMessage[i].Length;
                     }
-
-                    message = new Message(ReadableBuffer.Create(totalBuffer).Preserve(), Format.Text, receiveResult.EndOfMessage);
+                    
+                    message = new Message(ReadableBuffer.Create(totalBuffer).Preserve(), receiveResult.MessageType == 0 ? Format.Text : Format.Binary, receiveResult.EndOfMessage);
```

Comment:
Use WebSocketMessageType enum instead of 0. I think you are missing handling a case where the messge type is Close.